### DISCUSSION TITLE
Release v4.0.0

### DISCUSSION THREAD

**Discussion by smichel17** [LibreShift/red-moon#324] [General]
### Added
- Root Mode BETA (@notjuliee). BETA means it is not tested on many devices. Make sure you can reboot your phone without looking at the screen, just in case.
- Initial/incomplete support for Android 12 (@AdamNiederer).
- Start/stop times can be set to sunset/sunrise individually
- Settings screen with all non-profile preferences
- Hebrew translation (@Shopimisrel)

### Changed
- All non-filter preferences moved to their own screen
- Revert top switch toggle back to a floating button
- Prevent changing filter preferences until overlay permission is granted

### Fixed
- Time preference summary now displays in proper locale
- Fixed crash when toggling via the tile (@gilbsgilbs and @automaton82)
- Maybe fixed filter sometimes not turning on the very first time after installation

### Removed
- Removed "Sunrise to Sunset" preference, since you can set them individually
- Removed separate Google Play version (which included a prompt to rate the app)
- Removed Red Moon from the Google Play Store

<hr /><em>This discussion was created from the release <a href='https://github.com/LibreShift/red-moon/releases/tag/v4.0.0'>v4.0.0</a>.</em>

**Comment by sudo-py-dev** [comment]
You can update it on f-droid?

**Reply by smichel17** [reply]
Generally f-droid builds based on tags on this repo. I've already added the tag, so it will show up in f-droid whenever f-droid picks it up. Unless there are build errors or something.

**Reply by sudo-py-dev** [reply]
Okay good to know I didn't know how it works I'm just translating :)

**Reply by smichel17** [reply]
If it is not in f-droid by Sunday, I will go looking to see why it has not been built yet

**Reply by gilbsgilbs** [reply]
I think the problem is that the `versionCode` and `versionName` weren't updated accordingly in the `v4.0.0` tag: https://github.com/LibreShift/red-moon/blob/v4.0.0/app/build.gradle#L37-L38 . As a result, F-Droid auto-updater is unable to detect the v4.0.0 tag as containing a newer version, and therefore skips it. The [metadata](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/com.jmstudios.redmoon.yml) still doesn't include v4.0.0 although the auto-updater has run a few times since the tag was made, so no build is to be expected from the server for now.

**Reply by smichel17** [reply]
Ah, right! Forgot to bump that, it's been a while since I did an Android release :sweat_smile:

**Reply by smichel17** [reply]
I've updated them now and force-pushed to update the `v4.0.0` tag in git, we'll see if Red Moon picks that up; if not I'll tag a `v4.0.1` to fix it. Thanks @gilbsgilbs for the tip and @Shopimisrel for the translation :)

**Reply by gilbsgilbs** [reply]
> Ah, right! Forgot to bump that, it's been a while since I did an Android release sweat_smile

Those manual bumping steps are cumbersome and pretty painful in almost every languages and frameworks around there. There are often ways to automate release processes to prevent such oversights, but I'm unsure which options are available in Android projects as I don't work on them on a daily basis. And it might not be that relevant for projects that don't get released very often.

> we'll see if ~Red Moon~ F-Droid picks that up

I ran fdroid checkupdate on my machine and it was able to pick the update thanks to your fix, so we should see it appear [here](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/com.jmstudios.redmoon.yml#L249) tomorrow around 6:30 UTC.  🙂 

We'll then have to wait a bit for the build server to run. The build log will be available here: https://f-droid.org/repo/com.jmstudios.redmoon_39.log.gz . And then wait a few more days for the build to be signed and published by some F-Droid folk.

> Thanks @gilbsgilbs for the tip and @Shopimisrel for the translation :)

Thank YOU @smichel17 for your time and work on red-moon.

**Reply by gilbsgilbs** [reply]
FYI the build failed. I opened https://gitlab.com/fdroid/fdroiddata/-/merge_requests/11591 to fix it.

**Comment by xxxb-g** [comment]
I tried the root Mode on my Galaxy A145R (Android 14, magisk rooted) and it works really well. When I added this screenshot, I wanted to show it is working, but it is not affecting screenshots --> excellent work!

![Screenshot_20250905_214657_Fennec](https://github.com/user-attachments/assets/c59c0032-5c5f-40d2-8718-c9fedd5f4142)
